.github/workflows/checker.yml

```diff
@@ -0,0 +1,37 @@
+name: Checker
+
+on:
+  push:
+    branches: [ "dev" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
```

Comment:
**Suggestion:** Use matrix strategy for testing across multiple versions of python.
```suggestion
        strategy:
          matrix:
            python-version: ['3.8', '3.9', '3.10', '3.11']
        - name: Set up python
          id: setup-python
          uses: actions/setup-python@v4
          with:
            python-version: ${{ matrix.python-version }}
```